### PR TITLE
Add checkIdentityHealth api in Microsoft.Subscription.Admin RP.

### PR DESCRIPTION
**Pull request by yusonMicrosoft**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/azsadmin/resource-manager/subscriptions/Microsoft.Subscriptions.Admin/preview/2015-11-01/Subscriptions.json [hunk 1]
```diff
@@ -474,6 +504,240 @@
...
+                    "type": "string",
+                    "format": "date-time"
+                },
+                "reportEndTimeUtc": {
+                    "description": "End time of the report.",
+                    "type": "string",
+                    "format": "date-time"
+                },
+                "status": {
+                    "description": "The roll-up health status for the directory",
```
Comment (kpajdzik) [inline specification/azsadmin/resource-manager/subscriptions/Microsoft.Subscriptions.Admin/preview/2015-11-01/Subscriptions.json]:
```suggestion
                    "description": "The roll-up health status for the directory.",
```

File: specification/azsadmin/resource-manager/subscriptions/Microsoft.Subscriptions.Admin/preview/2015-11-01/Subscriptions.json [hunk 2]
```diff
@@ -474,6 +504,240 @@
                 }
             }
         },
+        "IdentityHealthCheckReportDefinition": {
```
Comment (kpajdzik) [inline specification/azsadmin/resource-manager/subscriptions/Microsoft.Subscriptions.Admin/preview/2015-11-01/Subscriptions.json]: Why not just `IdentityHealthCheckReport`?

File: specification/azsadmin/resource-manager/subscriptions/Microsoft.Subscriptions.Admin/preview/2015-11-01/Subscriptions.json [hunk 3]
```diff
@@ -474,6 +504,240 @@
...
+                "tenantId": {
+                    "description": "The tenant unique identifier.",
+                    "type": "string"
+                },
+                "tenantName": {
+                    "description": "The directory tenant resource name.",
+                    "type": "string"
+                },
+                "status": {
+                    "description": "The roll-up health status for the directory",
```
Comment (kpajdzik) [inline specification/azsadmin/resource-manager/subscriptions/Microsoft.Subscriptions.Admin/preview/2015-11-01/Subscriptions.json]:
```suggestion
                    "description": "The roll-up health status for the directory.",
```

File: specification/azsadmin/resource-manager/subscriptions/Microsoft.Subscriptions.Admin/preview/2015-11-01/Subscriptions.json (hunk 2 above)
Comment (yusonMicrosoft) [reply specification/azsadmin/resource-manager/subscriptions/Microsoft.Subscriptions.Admin/preview/2015-11-01/Subscriptions.json]: The 'IdentityHealthCheckReportDefinition' is consistent with the class name in source code. I followed the existing conventions like the 'CheckNameAvailabilityDefinition' in the Subscriptions.json.

File: specification/azsadmin/resource-manager/subscriptions/Microsoft.Subscriptions.Admin/preview/2015-11-01/Subscriptions.json (hunk 3 above)
Comment (yusonMicrosoft) [reply specification/azsadmin/resource-manager/subscriptions/Microsoft.Subscriptions.Admin/preview/2015-11-01/Subscriptions.json]: Done

File: specification/azsadmin/resource-manager/subscriptions/Microsoft.Subscriptions.Admin/preview/2015-11-01/Subscriptions.json (hunk 1 above)
Comment (yusonMicrosoft) [reply specification/azsadmin/resource-manager/subscriptions/Microsoft.Subscriptions.Admin/preview/2015-11-01/Subscriptions.json]: Done

**Review by kpajdzik — APPROVED** [review]

**Pull request merged**